### ISSUE TITLE
[BUG] playwright doesnt capture network response and returns error Error: response.json: Protocol error (Network.getResponseBody): No resource with given identifier found

### ISSUE THREAD

**Issue by InduKrish** [microsoft/playwright#26388]
### System info
- Playwright Version: 1.36.2
- Operating System: Mac
- Browser:All
- Other info:

### Source code

after clicking on the link ,want to capture the response it works most of the time and it fails sometimes. Please advise why it fails intermittently, why sometimes playwright is unable to capture the response,

it returns **Error: response.json: Protocol error (Network.getResponseBody): No resource with given identifier found**

when i test manually , after reload(), i see that all endpoints are loading as per the screenshot, however the test is not able to capture.

```javascript
const [auctionResponse] = await Promise.all([
            page.waitForResponse(res =>
                res.status() == 200
                &&
                res.url().includes('/vacation/auctions/')
                &&
                !res.url().includes('/settings')
               ),
            page.getByRole('link').getByText('2024').click(),
            page.reload(),
            page.waitForSelector('div[data-testid=\'wrapper-Bases\']')
        ]);
        const auctionResult = JSON.stringify(await auctionResponse.json())
        let accrualUrl = JSON.parse(auctionResult)._links.self.href;
        console.log("Accrual Url: " + accrualUrl)
        return auctionResponse;
```

error
```javascript
Error: response.json: Protocol error (Network.getResponseBody): No resource with given identifier found

  57 |         ]);
  58 |
> 59 |         const auctionResult = JSON.stringify(await auctionResponse.json())
     |                                                                    ^
  60 |         let accrualUrl = JSON.parse(auctionResult)._links.self.href;
  61 |         console.log("Accrual Url: " + accrualUrl)
  62 |         return auctionResponse;
```

Note:

before clicking on -2024 hyperlink, the URL is  https://service.corp.com/vacation/auctions

After i click on 2024 hyperlink on the page, it loads get url response, that i want to read api response from..,
the URL will add order id at the end (in the address bar)
 https://service.corp.com/vacation/auctions/**64e581be-4a7e-4f61-8390-b192d749865c**

and it is on the same URL, just the order id gets added to the end of the URL when i click on the 2024 hyperlink, thats where get API response gets loaded, and want to read the response. 

Intially without page.reload() also , i was getting the same error . i added page.reload() to ensure that api responses are loaded, hoping reload would fix the problem to reload all api responses.
Network.getResponseBody): No resource , thought page.reload would fix it, but it didnt. 
Both with and without page.reload i see getResponse body error. 
Please advise how to fix the issue? it only works sometimes.

**Other than clicking on 2024 and waiting for the response, i am not navigating to any other tab. for the vacation/auction response to load, i must click on 2024 hyperlink.**
As soon as i click on 2024, all the network requests that you see in the screenshot will load.  64e581be-4a7e-4f61-8390-b192d749865c is the network response i want to cpature (as listed in the screenshit)

<img width="1221" alt="Screen Shot 2023-08-09 at 4 11 37 PM" src="https://github.com/microsoft/playwright/assets/113629123/b8fe3c5c-0411-4d96-9ed9-1510d22fcb5b">

**Comment by dgozman** [comment]
@InduKrish This is because you are capturing the response, click and reloading at the same time. When the page navigates away, Playwright cannot retrieve the response from the browser, because it's already gone.

You should not reload right away, but instead click and capture the response first, and only then reload. Let me know whether that helps.

**Comment by InduKrish** [comment]
even when i dont reload(), i see the same error sometimes.

**Comment by dgozman** [comment]
@InduKrish Are you sure that `/vacation/auctions/` is triggered only after the click? Or perhaps there are many such requests, and you just capture a wrong one? I suspect that the network request you capture is sometimes issued before the click, and sometimes after, and when you capture the one before the click, it's body may be gone.

If you could provide a self-contained repro for us to run locally, we might be able to help better. Otherwise it's just random guesses.

**Comment by InduKrish** [comment]
ok, using !res.url().includes('/settings') - negation i am filtering out the url that i do not need, it is looking for the right /vacation/auctions/  that i verified.

Secondly, when i tested, /vacation/auctions/ is triggered before the click, and after the click as well it loads /vacation/auctions/ , and when i manually verify in network tab the first request before the click, i still see json response in the network tab, can you please explain what you mean by 'body may be gone'?


also, can you please clarify if playwright can capture the response although many other requests load after/vacation/auctions after the click? here after the click, vacation/auction api and many other api requests will load . I believe playwright will ignore all other requests and only look for url that matches /Vacation/auction can you please confirm?

**Comment by dgozman** [comment]
@InduKrish I won't be able to answer so many questions, but I can clarify a few things:

> Is playwright looking beyond/past the very first url and complaining with the error?

`waitForResponse` returns the first response that matches your criteria. It does not consider any other later responses that could also match. So if that particular response is gone, most likely because it happened before the navigation, you won't get the json. You should make sure you only match the last response you are interested in.

As I said before, if you could provide a self-contained repro for us to run locally, we might be able to help better. Otherwise it's just random guesses that are not really productive.

If you believe there is a bug in Playwright, please file a new issue with a self-contained repro by filling in the "Bug Report" template. Note that we should be able to run the repro locally.

**Comment by InduKrish** [comment]
i have seen the cases, where playwright is looking beyond the first response even after the URL is loaded right after the click, while using waitForResponse() , it actually read /setting url that also has url /vacation/auctions/setting, which is why i had to add !res.url().includes('/settings') -negation to ignore /settings url. 

since repo has confidential tokens, i am unable to share the repo for now.

**Comment by InduKrish** [comment]
if it the below (after click) is treated as a navigation, Can you please advise me how to handle that? I found the link below and the answer recommended.. 

I want to capture the response of http://aaa.corp.com/vacation/auction/64e581be-4a7e-4f61-8390-b192d749865c after click, not before click.
not sure if i  have to intercept the network response of the URL after Click, which is http://aaa.corp.com/vacation/auction/64e581be-4a7e-4f61-8390-b192d749865c?
Can you please advise how intercepting network helps here? and can you please share the example/reference if any?

eg: before click --> http://aaa.corp.com/vacation/auction
after click ----> http://aaa.corp.com/vacation/auction/64e581be-4a7e-4f61-8390-b192d749865c

https://github.com/microsoft/playwright-python/issues/1143


[yury-s](https://github.com/yury-s) commented [on Feb 1, 2022](https://github.com/microsoft/playwright-python/issues/1143#issuecomment-1027240159)
This is a known limitation, network responses are pruned on the next navigation. To work around that, you can try [intercepting](https://playwright.dev/python/docs/network#modify-requests) next navigation request and only [resume](https://playwright.dev/python/docs/api/class-route#route-continue) it once you got response body.

**Comment by mxschmitt** [comment]
We need a reproduction in order to assist here, these artificial URLs etc. don't help here much, unfortunately. Please re-file with a repro, since a new person will enter the triage shift today and this issue contains a lot of text already. Thanks for your understanding!
